apps/omg/test/omg/state/core_test.exs

```diff
@@ -911,6 +913,227 @@ defmodule OMG.State.CoreTest do
            )
   end
 
+  describe "Automatic fees claiming" do
+    setup do
+      fee_claimer = OMG.TestHelper.generate_entity()
+
+      {:ok, child_block_interval} = OMG.Eth.RootChain.get_child_block_interval()
+      {:ok, state_empty} = Core.extract_initial_state(0, child_block_interval, fee_claimer.addr)
+
+      alice = OMG.TestHelper.generate_entity()
+      fees = %{@eth => %{amount: 2}}
+
+      state =
+        state_empty
+        |> do_deposit(alice, %{amount: 10, currency: @eth, blknum: 1})
+        |> Core.exec(create_recovered([{1, 0, 0, alice}], @eth, [{alice, 8}]), fees)
+        |> success?()
+
+      {:ok, [state: state, alice: alice, fees: fees, fee_claimer: fee_claimer, state_empty: state_empty]}
+    end
+
+    test "should append fee txs in block", %{state: state} do
+      state = Core.claim_fees(state)
+      {:ok, {block, _dbupdates}, _state} = form_block_check(state)
+
+      assert [payment_txbytes, fee_txbytes] = block.transactions
+
+      %Transaction.Recovered{signed_tx: %Transaction.Signed{raw_tx: payment_tx}} =
+        Transaction.Recovered.recover_from!(payment_txbytes)
+
+      %Transaction.Recovered{signed_tx: %Transaction.Signed{raw_tx: fee_tx}} =
+        Transaction.Recovered.recover_from!(fee_txbytes)
+
+      assert %Transaction.Payment{} = payment_tx
+      assert %Transaction.Fee{} = fee_tx
+    end
+
+    test "fee txs are appended even when fees aren't required", %{state: state} do
+      state = Core.claim_fees(state)
+      {:ok, {block, _dbupdates}, _state} = form_block_check(state)
+
+      assert 2 = length(block.transactions)
+    end
+
+    test "should create utxos from claimed fee", %{alice: alice, state: state, fee_claimer: fee_claimer} do
+      state = Core.claim_fees(state)
+      {:ok, _, state} = form_block_check(state)
+
+      state
+      |> Core.exec(create_recovered([{1000, 1, 0, fee_claimer}], @eth, [{alice, 2}]), :no_fees_required)
+      |> success?()
+    end
+
+    test "fee txs cannot be intermixed with payments", %{alice: alice, state: state, fees: fees} do
+      fee_tx = create_recovered_fee_tx(1000, state.fee_claimer_address, @eth, 2)
```

Comment:
what do you mean by intermixed? that they need to be the last transactions in a block but not in the middle?

Reply:
Correct - fee tx by design should claim all surpluses from the current block, so no other payment txs can appear after.